dev-docs/source/GitConfig.rst

```diff
@@ -0,0 +1,168 @@
+.. _GitConfig:
+
+=================
+Mantid Git Config
+=================
+
+.. contents:: Contents
+   :local:
+
+Summary
+-------
+
+This page describes some configuration that will help with your usage of Git - some of which is strongly recommended. This should all be added to the ``.gitconfig`` file in your user home directory, i.e. ``~/`` on Linux or ``%USERPROFILE%`` on Windows. This will then be global across all of your repositories - please sanity check these suggestions if you use git for other repositories.
+
+Note that some of the values are commented out because they must be updated with your own details or are OS specific.
+
+Recommended Config
+------------------
+
+It is recommended to set up the following as a minimum:
+
+.. code-block:: sh
+
+   [user]
+           # Uncomment and set these to your own details
+           #name =
+           #email =
+   [core]
+           # Set an editor you like
+           editor = vi
+           # Specify your own global list of files/patterns that git should ignore
+           excludesfile = ~/.gitignore
+           # Fix up line endings between windows/linux - uncomment one
+           # Linux
+           #autocrlf = input
+           # Windows
+           #autocrlf = true
+   [log]
+           # Get the log to follow files even if they were renamed
+           follow = true
+   [fetch]
+           # Remove remote-tracking references that no longer exist on the remote when you do a fetch/pull
+           prune = true
+   [push]
+           # The safest default for git-push: push only the current branch and only if it has the same name as upstream
+           default = simple
+   [pull]
+           # Only pull the current branch by default
+           default = current
+           # Only allow pull if it can be fast-forwarded (otherwise you explicitly need to merge in the remote or rebase onto it e.g. with git pull --rebase)
+           ff = only
+   [diff]
+           # Add extra colours for highlighting moved code
+           colorMoved = default
+   [merge]
+           # Include the common ancestor in conflict details - very useful for working out which changes to keep
+           conflictstyle = diff3
+   [rebase]
+           # Automatically reposition !fixup/!squash commits in interactive rebase (useful with git commit --fixup/squash)
+           autosquash = true
+
+Diff/Merge Tools
+----------------
+
+You may find the following diff/merge tools helpful:
+
+.. code-block:: sh
+
+   # Uncomment lines where appropriate for specific OS
+   [diff]
+           # Set up a graphical tool for viewing diffs. I like meld because it looks good and does a good job with directory diffs.
```

Comment:
Can you add:
`but you'll need to install it separately`

Reply:
:+1: for Windows but I think meld is included on ubuntu?

Reply:
Can we just have some generic text that states third party tools need to be installed? Although, this does tread the line of obvious information.